village/src/App.js

```diff
@@ -11,14 +14,22 @@ class App extends Component {
       smurfs: [],
     };
   }
-  // add any needed code to ensure that the smurfs collection exists on state and it has data coming from the server
+
+  componentDidMount() {
+    axios.get('http://localhost:3333/smurfs')
+      .then(response => this.setState({smurfs: response.data}))
+      .catch(err => console.log(err));
+  }
+
+
   // Notice what your map function is looping over and returning inside of Smurfs.
   // You'll need to make sure you have the right properties on state and pass them down to props.
   render() {
     return (
       <div className="App">
-        <SmurfForm />
-        <Smurfs smurfs={this.state.smurfs} />
+        <Route path='/' component={NavBar} />
+        <Route path='/smurf-form' component={SmurfForm} />
```

Comment:
you probably wanna use render props and pass down a method so that your SmurfForm can tell App to update when needed